rest/utilities_testing.go

```diff
@@ -232,7 +240,36 @@ func (rt *RestTester) Bucket() base.Bucket {
 
 		rt.DatabaseConfig.SGReplicateEnabled = base.BoolPtr(rt.RestTesterConfig.sgReplicateEnabled)
 
-		_, err = rt.RestTesterServerContext.AddDatabaseFromConfig(*rt.DatabaseConfig)
+		if rt.leakyBucketConfig != nil {
+			// Scopes and collections have to be set on the bucket being passed in for the db to use.
+			if rt.createScopesAndCollections {
+				// WIP: Collections Phase 1 - Grab just one scope/collection from the defined set.
+				// Phase 2 (multi collection) means DatabaseContext needs a set of BucketSpec/Collections, not just one...
+				var scope, collection *string
+				for scopeName, scopeConfig := range rt.RestTesterConfig.DatabaseConfig.Scopes {
+					scope = &scopeName
+					for collectionName := range scopeConfig.Collections {
+						collection = &collectionName
+						break
+					}
+				}
+				if scope != nil && collection != nil {
+					collectionBucket, err := base.AsCollection(testBucket.Bucket)
+					if err != nil {
+						rt.tb.Fatalf("Could not get collection from bucket with type %T: %v", testBucket.Bucket, err)
+					}
+
+					collectionBucket.Spec.Scope = scope
+					collectionBucket.Spec.Collection = collection
+					collectionBucket.Collection = collectionBucket.Collection.Bucket().Scope(*scope).Collection(*collection)
```

Comment:
What exactly is going on here?  testBucket.Bucket is initialized as a gocb Collection targeting the default collection, and if we have a named collection we're using the associated Bucket() to swap to the underlying collection to the named collection?  
Are there any allocations associated with the default collection that need to be cleaned up?  I know Collection.Close() doesn't really do any work, but it feels a bit unusual to assign collectionBucket.Collection twice in this setup.

Reply:
There are no extra allocations done when changing the scope and collection to a named scope and collection. This is because the GoCB `Scope` and `Collection` functions return the provided bucket wrapped in a GoCB `Scope` and `Collection` struct.